### PR TITLE
Ngwmn 1759 - Authenticate using the Sass keycloak (Login.gov) within Django framework

### PR DESCRIPTION
**Pull request by mbucknell**
Before making a pull request
----------------------------

- [X] Make sure all tests run
- [X] Update the changelog appropriately

Description
-----------
This adds python-social-auth to the application to handle logging in via keycloak. It was tested using the WMA-B keycloak realm in sciencebase's keycloak. The login.gov integration is not working but I successfully logged in.

The social pipeline was modified to add code to automatically add usgs.gov users as staff so that they can access the admin. In addition they are added to the USGS group which allows access to edit the registry table. I added a custom setting SOCIAL_AUTH_DJANGO_SUPERUSERS which can be used to specify a list of email addresses of admin superusers who can access all tables. This can be used to add permissions to user's from other agencies so that they can edit their sites.
I will be adding at least a couple of tests for my custom social pipeline functions. I will also look into potentially testing the custom migration. 

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

**Comment by mbucknell** [comment]
Not sure why github is not picking up the travis build but it did work: https://travis-ci.org/github/ACWI-SOGW/well_registry_management/builds/695101048

File: wellregistry/registry/migrations/0002_add_agency_groups.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Custom data migration to add groups for each AGENCY defined in AGENCIES
+"""
+from django.db import migrations
+
+#pylint: disable=unused-argument
+#pylint: disable=invalid-name
+
+# Updating the values will cause the migration to be rerun
+AGENCIES = [
+    u'usgs',
```
Comment (ayan-usgs) [inline wellregistry/registry/migrations/0002_add_agency_groups.py:11]: `u` for unicode is not needed in Python 3.

File: wellregistry/wellregistry/custom_social_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+def change_usgs_user_to_staff(strategy, details, backend, *args, user=None, **kwargs):
+    """
+    Social auth pipeline function to change usgs users automatically to have is_staff set and
+    to add the usgs permissions to the user.
+    :return: dict
+    """
+    if kwargs.get('is_new'):
+        email = details.get('username')
+        if email.find('@usgs.gov') != -1:
```
Comment (ayan-usgs) [inline wellregistry/wellregistry/custom_social_pipeline.py:18]: Not a huge deal, but (subjectively) I think that that `'@usgs.gov' in email` is more. Additionally, using `in` is faster.

Comment (mbucknell) [reply wellregistry/wellregistry/custom_social_pipeline.py:18]: Agreed. I also am adding "@contractor.usgs.gov" here to check for.

File: Makefile [hunk 3]
```diff
@@ -21,7 +21,11 @@ test:
 
 runmigrations:
 	env/bin/python wellregistry/manage.py migrate --database=postgres postgres
-	env/bin/python wellregistry/manage.py migrate registry 0000
+	env/bin/python wellregistry/manage.py migrate admin
+	env/bin/python wellregistry/manage.py migrate auth
+	env/bin/python wellregistry/manage.py migrate contenttypes
+	env/bin/python wellregistry/manage.py migrate sessions
+	env/bin/python wellregistry/manage.py migrate social_django
 	env/bin/python wellregistry/manage.py migrate registry
```
Comment (duselmann) [inline Makefile:29]: unless the order is important, I think we can migrate in one command.
`env/bin/python wellregistry/manage.py migrate`

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,8 @@
-Django==3.0.6
+Django==3.0.7
 django-allow-cidr==0.3.1
 gunicorn==20.0.4
 pylint==2.5.2
 pylint-django==2.0.15
-python-dotenv==0.13.0
\ No newline at end of file
+python-dotenv==0.13.0
```
Comment (duselmann) [inline requirements.txt:6]: if we have a blank line at the end of the file then we will not get these false diffs.

File: Makefile (hunk 3 above)
Comment (mbucknell) [reply Makefile:29]: Good point. I was having issues earlier where it was helpful to have the order defined. I will test this out.

**Comment by duselmann** [comment]
This looks good. One more thing about migrations. Since we are moving to liquibase for the non-django database management. We should be able to no long store migrations in the repo. There is a makemigrations that can be run before migrate. The only reason I committed them was to manually manage the database.

**Comment by mbucknell** [comment]
We would still need to commit migrations (as recommended by Django). Migrations work like liquibase in that they check the migration database to see what has been run. If we run makemigrations it will make migrations for the current state of the models. Also I have already added one custom migration.

**Pull request merged**